### PR TITLE
FMWK-224 Add support for id projection query via Repository

### PR DESCRIPTION
**Pull request by agrgr**

### CODE REVIEW

File: src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java [hunk 1]
```diff
@@ -89,9 +95,26 @@ protected Query prepareQuery(Object[] parameters, ParametersParameterAccessor ac
...
+        if (queryMethod.getReturnedObjectType() != queryMethod.getEntityInformation().getJavaType()) {
+            return queryMethod.getReturnedObjectType();
+        }
+        // No projection - target class will be the entity class.
+        return queryMethod.getEntityInformation().getJavaType();
+    }
+
     public Query createQuery(ParametersParameterAccessor accessor, PartTree tree) {
         Constructor<? extends AbstractQueryCreator<?, ?>> constructor = ClassUtils
             .getConstructorIfAvailable(queryCreator, PartTree.class, ParameterAccessor.class);
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java:113]: Add validation
```java
Assert.notNull(constructor, "constructor must not be null");
```

File: src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java:1]: Lines 89-93 use
```java
if (query.getCriteria() instanceof SpelExpression spelExpression) {
    EvaluationContext context = this.evaluationContextProvider.getEvaluationContext(queryMethod.getParameters(),
        parameters);
    spelExpression.setEvaluationContext(context);
}
```

File: src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java [hunk 2]
```diff
@@ -46,6 +50,8 @@ protected BaseAerospikePartTreeQuery(QueryMethod queryMethod,
         this.queryMethod = queryMethod;
         this.evaluationContextProvider = evalContextProvider;
         this.queryCreator = queryCreator;
+        this.sourceClass = queryMethod.getEntityInformation().getJavaType();
+
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java:54]: Remove empty line

File: src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java [hunk 3]
```diff
@@ -89,9 +95,26 @@ protected Query prepareQuery(Object[] parameters, ParametersParameterAccessor ac
...
+    }
+
     public Query createQuery(ParametersParameterAccessor accessor, PartTree tree) {
         Constructor<? extends AbstractQueryCreator<?, ?>> constructor = ClassUtils
             .getConstructorIfAvailable(queryCreator, PartTree.class, ParameterAccessor.class);
         return (Query) BeanUtils.instantiateClass(constructor, tree, accessor).createQuery();
     }
+
+    protected static boolean isIdProjectionQuery(Class<?> targetClass, Object[] params, AerospikeCriteria criteria) {
+        return targetClass != null && params.length > 0 && Objects.equals(criteria.get(FIELD), "id");
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java:118]: `params` can be null here

File: src/main/java/org/springframework/data/aerospike/repository/query/AerospikePartTreeQuery.java [hunk 4]
```diff
@@ -52,11 +53,27 @@ public Object execute(Object[] parameters) {
 
         Class<?> targetClass = getTargetClass(accessor);
 
+        if (isIdProjectionQuery(targetClass, parameters, (AerospikeCriteria) query.getCriteria())) {
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/AerospikePartTreeQuery.java:56]: * Add a comment explaining this block
* Is `query.getCriteria()` always an AerospikeCriteria?

File: src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java [hunk 5]
```diff
@@ -42,20 +44,24 @@ public Object execute(Object[] parameters) {
...
         Query query = prepareQuery(parameters, accessor);
         Class<?> targetClass = getTargetClass(accessor);
-        return findByQuery(query, targetClass);
-    }
 
-    private Class<?> getTargetClass(ParametersParameterAccessor accessor) {
-        // Dynamic projection
-        if (accessor.getParameters().hasDynamicProjection()) {
-            return accessor.findDynamicProjection();
+        if (isIdProjectionQuery(targetClass, parameters, (AerospikeCriteria) query.getCriteria())) {
```
Comment (reugn) [inline src/main/java/org/springframework/data/aerospike/repository/query/ReactiveAerospikePartTreeQuery.java:48]: Add a comment explaining this block

**Review by reugn — APPROVED** [review]

File: src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java (hunk 1 above)
Comment (agrgr) [reply src/main/java/org/springframework/data/aerospike/repository/query/BaseAerospikePartTreeQuery.java:112]: The same assertion is the first line inside BeanUtils.instantiateClass()

**Pull request merged**